src/main/java/org/haffson/adventofcode/utils/FileReaders.java

```diff
@@ -14,4 +13,18 @@
 @Component
 public class FileReaders {
 
+    public int[] readFileIntoArrayOfIntegers(String path){
+        List<String> data = new ArrayList<>();
+        InputStream inputStream = FileReaders.class.getResourceAsStream(path);
+        Scanner scanner = new Scanner(inputStream);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            if(!line.isEmpty()){
+              data.add(line);
+            }
+        }
+        scanner.close();
```

Comment:
By using try-with-resources statements, you can ensure that resources are **always** released. 
Even in case of exceptions:

```
try (Scanner scanner = new Scanner(inputStream) {
        while (scanner.hasNextLine()) {
               // ...
    }
}
```

try-with-resources  makes manual ``close()`` calls obsolete.
``close()`` will be called automatically, even if an exception is thrown within the ``try`` block.

Reply:
Thanks for the tip, I will change the code section.